extensions/webirc-3.3.md

```diff
@@ -0,0 +1,65 @@
+---
+title: IRCv3 WebIRC extension
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Evan Magaliff"
+    period: "2017"
+    email: "muffinmedic@kiwiirc.com"
+  -
+    name: "Darren Whitlen"
+    period: "2017"
+    email: "prawnsalad@kiwiirc.com"
+---
+## Purpose
+This specification is intended to document the existing `WEBIRC` command that is already in use by multiple IRCDs and WebIRC gateways to provide a standard specification for implementation. This specification does not add any new features or changes, although new features, such as use of public-key cryptography over password authentication, can be submitted as extensions.
+
+## Description
+When a user connects through an indirect connection to the IRC server, the user's actual IP address and hostname are not visible. The `WEBIRC` command allows the WebIRC gateway to pass on the user's actual IP address and hostname to the IRC server. This information can be utilized by the network as if the user was connecting directly.
+
+## Format
+The `WEBIRC` command takes four parameters: `pass `, `user`, `hostname`, and `ip`. `pass` is the password authenticating the WebIRC gateway to the IRC server which must be agreed upon ahead of time, `user` is the WebIRC gateway service name, `hostname` is the resolved hostname of the user's IP address, and `ip` is the IPv4 or IPv6 address of the connecting user (IPv4-in-IPv6 MUST NOT be sent). If the IP address cannot be resolved to a DNS name, the IP address MUST be sent as both the `hostname` and `ip`.
```

Comment:
Also, is there any reason for specifying this:

>IPv4-in-IPv6 MUST NOT be sent

Seems like its something that should be left up to the implementations to decide on how they want to handle that.